FEATURE: make media browser uri path configurable

so-grimm opened the pull request:
- add and use new data-attribute and const for mediaBrowserUriPath with fallback value of '/neos/media/browser'
- add explanation for new data-attribute to Readme

Todo:
- [x]  Minify js

so-grimm committed:
TASK: make media browser uri path configurable

- add and use new data-attribute and const for mediaBrowserUriPath with fallback value of '/neos/media/browser'
- add explanation for new data-attribute to Readme

ahaeslich commented on Resources/Private/source.js:
```diff
@@ -135,7 +135,8 @@ window.addEventListener('DOMContentLoaded', () => {
 		const assetIdentifier = container.dataset.asset;
-		const url = assetIdentifier && !multiple ? new URL('/neos/media/browser/images/edit.html?asset=' + assetIdentifier, window.location.origin) : new URL('/neos/media/browser/assets/index.html', window.location.origin);
+		const mediaBrowserUriPath = container.dataset.assetMediaBrowserUriPath || '/neos/media/browser';
+		const url = assetIdentifier && !multiple ? new URL(mediaBrowserUriPath + '/images/edit.html?asset=' + assetIdentifier, window.location.origin) : new URL(mediaBrowserUriPath + '/assets/index.html', window.location.origin);
```
Looking at the code of this plugin: If we respect the (potential) subfolder here. Isn't it missing in line 39?

bwaidelich replied:
The comment is now marked outdated but there are only two commits from 1 h ago!? @so-grimm did you force-push to the branch?

ahaeslich replied:
This comment was made on the first commit. The second commit did change these lines ... Shouldn't they be outdated now?

bwaidelich replied:
yes, you're right of course. I misinterpreted the commit log

so-grimm committed:
TASK: use new data attribute for preview

- change name of data attribute to more general one
- use it for preview as well

bwaidelich committed:
Tweak README and minify JS

bwaidelich commented:
@so-grimm Thanks for this – I tried to push a tweak to the README and the minified JS but I'm not allowed to. Could you "Allow edits by maintainers"?

bwaidelich commented on README.md:
```diff
@@ -36,6 +36,7 @@ Neos Media Browser and/or asset preview:
 | `asset-constraints-media-types`   | Comma separated list of allowed mediaTypes (optional)                                                   | container                                                         |
 | `asset-constraints-asset-sources` | Comma separated list of allowed asset sources (optional)                                                | container                                                         |
 | `asset-multiple`                  | If set, multiple assets can be added (optional)                                                         | container                                                         |
+| `asset-neos-uri-path`             | Will change the uri path. I.e. necessary when working with subfolders                                   | container                                                         |
```
```suggestion
| `asset-neos-uri-path`             | Can be set if the Neos routes don't reside underneath `/neos` (optional)                                | container                                                         |
```

bwaidelich reviewed (commented):
I added my suggestions as inline suggestion now.

For the `dist.js` you can just use https://www.toptal.com/developers/javascript-minifier and past the source code of `dialog-polyfill.js` and `source.js`

ahaeslich committed:
Update README.md

Co-authored-by: Bastian Waidelich <b.waidelich@wwwision.de>

so-grimm commented:
@bwaidelich I just found out that it's actually not possible to "Allow edits by maintainers" if the PR is coming from an organisation and not an individual. But Anke gave you write-access to the repo, so you should be able to fix whatever you need to now.

bwaidelich committed:
Merge branch 'task/configurable-media-browser-uri' of https://github.com/queoGmbH/Wwwision.Neos.MediaBrowserDialog into task/configurable-media-browser-uri

bwaidelich commented:
@so-grimm @ahaeslich Thanks a lot!
=> https://github.com/bwaidelich/Wwwision.Neos.MediaBrowserDialog/releases/tag/2.4.0
